pkg/cstorpool/v1alpha2/cstorpool_test.go

```diff
@@ -0,0 +1,167 @@
+package v1alpha2
+
+import (
+	"testing"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestAll(t *testing.T) {
+	tests := map[string]struct {
+		mockPredicates predicateList
+		expectedOutput bool
+	}{
+		// Positive predicates
+		"Positive Predicate 1": {[]predicate{func(*csp) bool { return true }}, true},
+		"Positive Predicate 2": {[]predicate{func(*csp) bool { return true }, func(*csp) bool { return true }}, true},
+		"Positive Predicate 3": {[]predicate{func(*csp) bool { return true }, func(*csp) bool { return true }, func(*csp) bool { return true }}, true},
+		// Negative Predicates
+		"Negative Predicate 1":  {[]predicate{func(*csp) bool { return false }}, false},
+		"Negative Predicate 2":  {[]predicate{func(*csp) bool { return true }, func(*csp) bool { return false }}, false},
+		"Negative Predicate 3":  {[]predicate{func(*csp) bool { return true }, func(*csp) bool { return false }}, false},
+		"Negative Predicate 4":  {[]predicate{func(*csp) bool { return false }, func(*csp) bool { return true }}, false},
+		"Negative Predicate 5":  {[]predicate{func(*csp) bool { return false }, func(*csp) bool { return false }}, false},
+		"Negative Predicate 6":  {[]predicate{func(*csp) bool { return false }, func(*csp) bool { return true }, func(*csp) bool { return true }}, false},
+		"Negative Predicate 7":  {[]predicate{func(*csp) bool { return true }, func(*csp) bool { return false }, func(*csp) bool { return true }}, false},
+		"Negative Predicate 8":  {[]predicate{func(*csp) bool { return true }, func(*csp) bool { return true }, func(*csp) bool { return false }}, false},
+		"Negative Predicate 9":  {[]predicate{func(*csp) bool { return true }, func(*csp) bool { return false }, func(*csp) bool { return false }}, false},
+		"Negative Predicate 10": {[]predicate{func(*csp) bool { return false }, func(*csp) bool { return false }, func(*csp) bool { return false }}, false},
+	}
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			if output := mock.mockPredicates.all(&csp{}); output != mock.expectedOutput {
+				t.Fatalf("test %q failed: expected %v \n got : %v \n", name, mock.expectedOutput, output)
+			}
+		})
+	}
+}
+
+func TestIsNotUID(t *testing.T) {
+	tests := map[string]struct {
+		mockcsp        *csp
+		uids           []string
+		expectedOutput bool
+	}{
+		// Positive Test
+		"Positive 1": {&csp{&apis.CStorPool{ObjectMeta: metav1.ObjectMeta{UID: "uid6"}}}, []string{"uid1", "uid2", "uid3", "uid4"}, true},
```

Comment:
Remove the redundant struct construction logic into the code before assertion etc is done.
The table here should consists of primitive data types i.e. int, string, array of int or array of string, predicate function names.

Reply:
done

Reply:
done